CONTRIBUTING.md

```diff
@@ -1,81 +1,132 @@
+  maintainability of the code. If your change is Elo-positive but makes the code
+  more complex, please discuss it first. Recently, we added an option to
+  clone the search algorithm in extreme cases.
+* Elo-neutral simplifications are always welcome.
+
+Pull Requests are squashed when merged. This means all commits in the branch
+will be squashed into one commit applied onto master. This makes it tricky to
+reuse the branch and continue to work on it after the PR is merged.
+
+**Note:** This section only applies if you have dependent branches that were
+built on top of your merged PR branch. If you only had one branch, you can
+simply delete it after merging.
+
+**Example scenario:** You had a branch `add-feature` that got merged, and you
+have another branch `extend-feature` that was based on `add-feature`. After
+`add-feature` is merged, you need to rebase `extend-feature` onto the new
+master. Here's what to do after your PR is merged:
+
+```shell
+git fetch upstream  # Update your local master
+git checkout extend-feature  # Switch to your dependent branch
+git rebase --update-refs --onto upstream/master add-feature # Rebase onto the updated master
+```
+
+The `--update-refs` flag will also update any branches between your leaf branch
+and the merged branch if you have any.
+
+## C++ Standard and Libraries
+
+* We use most C++20 features. However, supported compilers are GCC 10 and clang
+  10, so some features may not be available.
+* We use protocol buffers. However, we don't use any external library for it,
+  but rather generate the code from `.proto` files using the script in
+  `scripts/`.
+* Since v0.32, we use Abseil (`absl::`).
+* Use `CERR` for logging (goes to stderr and log), or `LOGFILE` (goes to log
+  file only).
+* Writing tests is encouraged. We use `gtest`/`gmock` for unit tests. Tests are
+  located in the same directory as the code they test, in a file with the same
+  name but ending with `_test.cc`.
+
+## Style Guidelines
+
+We follow the
+[Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html) with
+these modifications:
+
+* **Header guards**: Use `#pragma once` instead of traditional header guards.
+* **Exceptions**: are allowed, but only one: `lczero::Exception`.
+* **References vs Pointers**: Non-const reference parameters are neither
+  encouraged nor discouraged over pointers (in the Google style guide, they used
+  to be discouraged, and now they are encouraged).
+* **Formatting**: Run `clang-format` on all code before committing.
+* **RValue references**: Rvalue reference function parameters are allowed, not
+  only for constructors and assignment operators. However, use them only if they
+  benefit readability or have significant performance gain.
+* Every new file should contain a GPLv3 banner with:
+  * Additional exception under GNU GPL version 3 section 7 regarding NVIDIA
+    libraries (see examples in other files).
```

Comment:
We also allow backends to be GPL only without the exception..

Reply:
Updated.